### DISCUSSION TITLE
关于API文档的自动生成方案

### DISCUSSION THREAD

**Discussion by rfvtgbzxc** [rfvtgbzxc/RMMZ-TS-Vite#1] [Ideas]
本项目意图尽量降低RM插件开发的门槛，为此已经设计了一些简单的功能封装API，后续还会追加。为了方便对这些API进行查阅，考虑采用某种方案对这些API生成文档。
本讨论还讨论关于monorepo模式的应用。由于本项目是一个脚手架+基于脚手架的插件范例的开发模式，因此的确存在脚手架频繁更新后，插件范例的对应脚手架也要更新的情况，monorepo似乎可以将两种情况下的脚手架整合到一起，具有实践价值。

以下是一份生成方案的初步讨论文章，from @RuanZhongNan：
# to rfvtgbzxc

一些答疑。

## 针对项目

- https://github.com/rfvtgbzxc/RMMZ-TS-Vite

## 生成的文档是针对脚手架的还是插件的？

同时生成。两个都可以做。

其本质是 vuepress 配置文件，扫描项目的目录结构，然后生成对应的文档。所以生成的文档可以针对`脚手架`，也可以针对`插件`。

个人建议是先实现对手脚架的文档生成，再实现针对插件的文档生成。

## 可以自定义参考信息？

我不明白你指的`参考信息`是什么。
- re：说重复了，就是指可以自定义从脚手架or插件，乃至更具体的单个文章去生成。

## 为什么文档也要放在 monorepo？

绝大多数项目都是这样做的。代码放在一侧，文档放在另一侧。

特别是自己写的文档，都是一起放在 monorepo 内的。

## 自动生成的 md 文件不参与 git 管理，不上传到仓库

不管我们使用的是 jsdoc 还是 typedoc，生成的文档都是 md 文件。基于代码文件生成的 md 文档，不参与管理。

## 生成文档的自动工作流？

这个随意。

- github workflow
- vercel
- gulp

对此我暂时没有好的实践方案。我还没有涉及到`自动部署+发包+部署文档`的内容。但是可以肯定的是，生成文档肯定是一个命令。这个命令可以和部署命令相结合。也可以用 gulp 统一配置上述的工作任务。

大概是这样的：

- jsdoc/typedoc 扫描文件生成 md
- vuepress2 扫描刚刚生成的 md，和已经写好的 md，共同生成出静态网页。存放在某个 dist 内。
- vite 手脚架扫描 rmmv/mz 模板。
- 手脚架编译 typescript 并部署到指定路径。

这样描述确实相当抽象，不够具体。我最好是给出一个小案例为好，不过我可能没空去弄这个。😂

# 反馈
感谢解答。
- monorepo的权限粒度相对较大，脚手架、示例部分的代码、文档，对于所有授权者都可以修改。这个倒不一定算问题，而且似乎专门的monorepo框架也提供了这个功能，但是reviewer的设置会因此更谨慎一些。
- API文档要更详细描述参数信息，需要写tsdoc，开发工作量上升，不过的确符合项目目的。会尝试先生成API文档，再逐渐追加tsdoc。
- monorepo这边，需要再阅读一下monorepo的实践方式，了解如何在保持脚手架和插件代码之间相对独立的情况下，插件代码又能引用脚手架进行正常开发。

**Comment by ruan-cat** [comment]
我已经会monorepo和typescript自动生成文档了。你人在哪里，我要狠狠pr。
